### PR TITLE
neonvm-kernel: Switch to version 6.6.63

### PR DESCRIPTION
**Pull request by petuhovskiy**
This commit updates the kernel and switches from 6.1.x to 6.6.x LTS branch.

As of 2024-12-06, www.kernel.org says this is the most recent version of the 6.6.x LTS branch.

Basically just followed the instructions in neonvm-kernel/README.md, copying in linux-config-ARCH-6.6.63 as .config, and .config back to /host once done with menuconfig. Fixed the instruction for amd64/arm64 along the way.

Trying this as a possible solution to https://github.com/neondatabase/cloud/issues/20195

### CODE REVIEW

**Review by sharnoff — COMMENTED** [review]
Seems fair enough, but I'll leave it to Misha to give a more proper review.

One question, one request.

**Question:** What's the thinking between upgrading to latest 6.1.x vs 6.6.x ?

**Request:** Due to changes in upgrade instructions — if you haven't already, could you double-check that going through the upgrade steps for the current version produces an empty diff? (and if not, let's separate that into its own PR?)

I ask in part because there may be some changes due to #1159, and if there's a bunch it'd be good to isolate those out so they can be ignored.

**Comment by petuhovskiy** [comment]
> Question: What's the thinking between upgrading to latest 6.1.x vs 6.6.x ?

Not much, just noticed that on kernel bugzilla it was a common suggestion to switch major version in similar cases:
- https://bugzilla.kernel.org/show_bug.cgi?id=218818#c4 – suggested to switch from 6.8.x to 6.6.x
- https://bugzilla.kernel.org/show_bug.cgi?id=218636#c1 – suggested to switch from 6.1.x to 6.7.x

**Comment by sharnoff** [comment]
> just noticed that on kernel bugzilla it was a common suggestion to switch major version in similar cases

fair enough, makes sense!

File: neonvm-kernel/README.md [hunk 1]
```diff
@@ -27,28 +29,54 @@ with the following sequence of actions:
...
-   rm linux-config-6.1.92
+   rm linux-config-amd64-6.6.63
    ```
 
 Afterwards, it's probably also good to do a search-and-replace repo-wide to update all places that
 mention the old kernel version.
 
+### On arm64 (aarch64 ARM)
+
+Note: you need to run this on ARM64 machine. It worked on my MacBook M2, but didn't on x64.
```
[the remark this answers is not in the document]
Comment (mikhail-sakhnov) [reply neonvm-kernel/README.md:50]: @sharnoff I answered in the pr #1172, should be doable cross platform

**Comment by petuhovskiy** [comment]
There is a new kernel version, so I opened a new PR to upgrade to 6.6.64
=> https://github.com/neondatabase/autoscaling/pull/1174

**Pull request closed without merging**